WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -528,6 +528,9 @@ private extension ProductFormViewController {
     }
 
     func onImageStatusesUpdated(statuses: [ProductImageStatus]) {
+        tableViewModel = DefaultProductFormTableViewModel(product: product,
```

Comment:
Due to the complicated cases of product editing, I think it's best to add clear comments on why we added this line. Maybe we should explain here that we need to update the view model to make sure the displayed product is update-to-date at this point?